### DISCUSSION TITLE
Resolving fetch loops

### DISCUSSION THREAD

**Discussion by zibonbadi** [zibonbadi/liquidms#13] [General]
The original master server API does not consider use of multiple servers, much less as a distributed, intercommunicating network.

This thread serves as discussion to the approach on how to resolve fetch loops within a network.

**Comment by zibonbadi** [comment]
I recommend using a regular polling loop of regular intervals instead of a fetch-on-request approach. This resolves the need to recursively fetch servers by only returning "flat" server lists per node at the expense of potentially slower updates of external server information. Eventual consistency should still be kept this way, however.

**Comment by thewane** [comment]
If we're going by polling loops we can use cron jobs to run the PHP functions, write them to be more CLI-friendly (I've done that in https://github.com/zibonbadi/liquidms/pull/4/commits/f7adc81f43a88ed7c84ab54d3187ee24c6d113e8 ), and add an output location to the .yaml for the rooms/servers files to get updated and pushed to whatever web server one could ask for

**Comment by zibonbadi** [comment]
Late reply but the addition of the fetch-from-snitch feature (and accompanying Snitch API) further "flattens" the data by providing universewide netgame data alongside it's origin, resolving origin indirection. By extension, it is highly recommended to fetch-from-snitch over relying on the v1 API wherever possible.
